src/main/java/org/jenkinsci/plugins/github_branch_source/RepositoryUriResolver.java

```diff
@@ -39,14 +35,13 @@
     public abstract String getRepositoryUri(String apiUri, String owner, String repository);
 
     public static String hostnameFromApiUri(String apiUri) {
-        apiUri = Util.fixEmptyAndTrim(apiUri);
-        try {
-            URL endpoint = new URL(apiUri);
-            if (endpoint.getPath().startsWith("/api/v3")) {
+        if (apiUri != null) {
+            try {
+                URL endpoint = new URL(apiUri);
                 return endpoint.getHost();
+            } catch (MalformedURLException e) {
+                // ignore
```

Comment:
So, if `apiUri` is not a well formed URL then `github.com` is returned. It's a weird behaviour from a UX point of view IMO (if the user sets something wrong then it must be notified back somehow).